fish.py

```diff
@@ -425,19 +437,28 @@ def blowcrypt_pack(msg, cipher):
     return '+OK ' + cipher.encrypt(padto(msg, 8))
 
 
-def blowcrypt_unpack(msg, cipher):
-    """."""
+def blowcrypt_unpack(msg, cipher, wrong_key_type=False):
+    """
+    Returns: str(message), bool(broken), bool(wrong_key_type)
+    Returns
+        utf-8 encoded decrypted string
+        boolean flag indicating whether the original ciphertext was broken or not
+        boolean flag returns True when a message prefix does not match the key type prefix
+    """
     if not (msg.startswith('+OK ') or msg.startswith('mcps ')):
         raise ValueError
     _, rest = msg.split(' ', 1)
 
     if rest[0] == "*":
         if cipher.mode != Blowfish.MODE_CBC:
-            raise ValueError
+            if not bool(cipher.key.decode("utf-8").startswith('cbc:')):
```

Comment:
```suggestion
            if not cipher.key.decode("utf-8").startswith('cbc:'):
```
as `startswith` already yields a bool anyway (https://docs.python.org/3/library/stdtypes.html#str.startswith)